File: cs/Iris/ExpressionElement/LabelExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Iris.ExpressionElement
{
    public class LabelExpression : Expression, IBlockExpression
    {

        public string labelName;
        public Expression body;

        public LabelExpression(string lName)
        {
            this.labelName = lName;
        }

        public LabelExpression(string lName, Expression lBody)
        {
            this.labelName = lName;
            this.body = lBody;
        }

        public List<string> declaredVariables()
        {
            return null;
        }
    }
}

File: java/Iris/src/com/rokuan/iris/expression/LabelExpression.java
package com.rokuan.iris.expression;

import java.util.ArrayList;

public class LabelExpression extends Expression implements IBlockExpression
{
	public String labelName;
	public Expression body;

	public LabelExpression(String lName)
	{
		this.labelName = lName;
	}

	public LabelExpression(String lName, Expression lBody)
	{
		this.labelName = lName;
		this.body = lBody;
	}

	public ArrayList<String> declaredVariables()
	{
		return null;
	}

	/*public override string ToString(string alinea)
        {
            StringBuilder str = new StringBuilder();

            str.Append(alinea);
            str.Append(labelName);
            str.Append(":\n");

            if (body != null)
            {
                str.Append(body.ToString(alinea + '\t'));
                str.Append('\n');
            }

            return str.ToString();
        }*/
}
